src/object/auto/autoegg.cpp

```diff
@@ -194,7 +195,9 @@ bool CAutoEgg::EventProcess(const Event &event)
 
             CProgramStorageObject* programStorage = dynamic_cast<CProgramStorageObject*>(alien);
             Program* program = programStorage->AddProgram();
-            programStorage->ReadProgram(program, m_alienProgramName.c_str());
+            programStorage->ReadProgram(program, InjectLevelPathsForCurrentLevel(m_alienProgramName, "ai"));
```

Comment:
There is similar code for ```produce()``` and ```BotFactory```. This code path is marked by ```program->readOnly = true;``` and is responsible for using the original file instead of saving a copy in ```savegame/Player/``` folder.

Code for [saving](https://github.com/colobot/colobot/blob/dev/src/object/implementation/program_storage_impl.cpp#L323) and [loading](https://github.com/colobot/colobot/blob/dev/src/object/implementation/program_storage_impl.cpp#L263) programs expect that ```struct Program``` will have the original string in its ```filename``` field, and if the filename field doesn't have ```%lvl%, %cat%, etc.```, the program loading code adds "ai/", where if the resolved path was saved, the game may try to load a non-existing file.
E.g. ```"ai/ai/something.txt"``` or ```"ai/<path from %lvl%>/something.txt"```.

Currently, we keep the original string and extract the real path at the point of use.
It might be possible to change the code to resolve paths at the beginning of the chain.

I think the current approach has some benefits;
* Corresponding strings in ```scene.txt```, ```data.sav```, and ```struct Program``` are always identical.
* The ```data.sav``` file is smaller and easier to read than it would be otherwise.
* It seems like this made it easier to find and fix this bug, and other similar bugs.